tests/beaker_tests/vni/test_vni.rb

```diff
@@ -0,0 +1,197 @@
+#
+# tests[id][:manifest_props] - This is essentially a master list of properties
+#   that permits re-use of the properties for both :present and :absent testing
+#   without destroying the list
+# tests[id][:resource_props] - This is essentially a master hash of properties
+#   that permits re-use of the properties for both :present and :absent testing
+#   without destroying the hash
+# tests[id][:title_pattern] - (Optional) defines the manifest title.
+#   Can be used with :af for mixed title/af testing. If mixing, :af values will
+#   be merged with title values and override any duplicates. If omitted,
+#   :title_pattern will be set to 'id'.
+#
+
+tests['preclean'] = {
+  title_pattern:  '10000',
+  manifest_props: '',
+  code:           [0, 2],
+  resource_props: {},
+}
+
+# Default property testing doesn't apply here. mapped_vlan => 'default'
+# amounts to nothing as there is no default vlan to vni mapping
+
+tests['non_default_properties'] = {
+  title_pattern:  '10000',
+  manifest_props: "
+    mapped_vlan     => '100',
+  ",
+  resource_props: {
+    'mapped_vlan' => '100'
+  },
+}
+
+#################################################################
+# HELPER FUNCTIONS
+#################################################################
+
+# Full command string for puppet resource command
+def puppet_resource_cmd
+  cmd = UtilityLib::PUPPET_BINPATH + 'resource cisco_vni'
+  UtilityLib.get_namespace_cmd(agent, cmd, options)
+end
+
+def build_manifest_vni(tests, id)
+  if tests[id][:ensure] == :absent
+    state = 'ensure => absent,'
+    tests[id][:resource] = {}
+  else
+    state = 'ensure => present,'
+    manifest = tests[id][:manifest_props]
+    tests[id][:resource] = tests[id][:resource_props]
+  end
+
+  tests[id][:title_pattern] = id if tests[id][:title_pattern].nil?
+  logger.debug("build_manifest_vni :: title_pattern:\n" +
+               tests[id][:title_pattern])
+  tests[id][:manifest] = "cat <<EOF >#{UtilityLib::PUPPETMASTER_MANIFESTPATH}
+  node 'default' {
+    cisco_vni { '#{tests[id][:title_pattern]}':
+      #{state}
+      #{manifest}
+    }
+  }
+EOF"
+end
+
+def test_harness_vni(tests, id)
+  tests[id][:ensure] = :present if tests[id][:ensure].nil?
+  tests[id][:resource_cmd] = puppet_resource_cmd
+  tests[id][:desc] += " [ensure => #{tests[id][:ensure]}]"
+
+  # Build the manifest for this test
+  build_manifest_vni(tests, id)
+
+  # For full test support of properties use test_harness_common; as an
+  # alternative use direct calls to individual test_* wrapper methods:
+  # test_harness_common(tests, id)
+  test_manifest(tests, id)
+  test_resource(tests, id)
+  test_idempotence(tests, id)
+
+  tests[id][:ensure] = nil
+end
+
+#################################################################
+# TEST CASE EXECUTION
+#################################################################
+test_name "TestCase :: #{testheader}" do
+  # -------------
+  id = 'preclean'
```

Comment:
Using the latest `develop` code you can now use the `resource_absent_cleanup()` api instead.

https://github.com/cisco/cisco-network-puppet-module/blob/develop/tests/beaker_tests/cisco_aaa_login_cfg_svc/test_aaalogincfgsvc_defaults.rb#L90